pytask_io/pytask_io.py

```diff
@@ -105,6 +75,81 @@ def __init__(self, *args, **kwargs):
         self.store_db = kwargs.get("store_db") or self.store_db
         self.workers = kwargs.get("workers") or self.workers
 
+        self._queue_client = None
+        self._queue_store = None
+        self._loop_thread = None
+        self._main_loop = None
+    
+    @property
+    def queue_client(self) -> redis.Redis:
+        """
+        The connected queue client object. Use this object exactly as you would 
+        if you were referencing the queue's client directly. Example::
+        
+            # Example for default Redis queue pushing a task into the queue
+            pytaskio = PytaskIO()
+            results = pytaskio.queue_client.lpush("my_queue", my_task)
+        
+        :raises NotReadyException: Raised when trying to a property before `run()` is called.
+        :return: :class:`redis.Redis` client object.
+        :rtype: redis.Redis
+        """
+        if not self._queue_client:
+            raise NotReadyException()
+        return self._queue_client
+    
+    @property
+    def queue_store(self) -> redis.Redis:
+        """
+        The `queue_store` is available to work with & can be accessed on the PyTaskIO instance
+        & all the available methods from the store framework used will be available on this object.
+        Foe example::
+        
+           # Example fpr default Redis store setting a new key
+           pytaskio = PytaskIO()
+           pytaskio.queue_store.set('myfield', 'my_value')
+
+        :raises NotReadyException: Raised when trying to a property before `run()` is called.
+        :return: :class:`redis.Redis` store object.
+        :rtype: redis.Redis
+        """
+        if not self._queue_store:
+            raise NotReadyException()
+        return self._queue_store
+
+    @property
+    def loop_thread(self) -> Thread:
+        """
+        The thread that the asyncio event loop runs in. This thread has been tagged with the name of
+        `event_loop`. The thread will die gracefully when PytaskIO calls `event_loop.join()` for you. If
+        you require custom handling of the `event_loop` thread, then you can access it directly using
+        the :class:`pytask_io.loop_thread` object. Example::
+        
+           pytaskio = PytaskIO()
+           pytaskio.loop_thread.is_alive() # check if the thread is still alive
+        
+        :raises NotReadyException: Raised when trying to a property before `run()` is called.
+        :return: :class:`Thread` object.
+        :rtype: threading.Thread
+        """
+        if not self._loop_thread:
+            raise NotReadyException()
+        return self._loop_thread
+    
+    @property
+    def main_loop(self) -> asyncio.AbstractEventLoop:
```

Comment:
This is my committed code, why have you overridden this line?

Reply:
Not sure what you mean by that, the only thing I copied is your docstring for the method, it would look nice if it was duplicated, would it?